src/osemgrep/cli/CLI.ml

```diff
@@ -276,5 +276,6 @@ let main argv : Exit_code.t =
   *)
   (*TOADAPT? adapt more of Common.boilerplate? *)
   let exit_code = safe_run ~debug (fun () -> dispatch_subcommand argv) in
+  Metrics_.add_exit_code exit_code;
   before_exit ~profile ();
```

Comment:
I guess we need to send the payload at some point before exiting.

Reply:
maybe add a giant TODO comment, explaining what remains to be done here.

Reply:
Actually, if only the `scan` sub-command matters, we probably can add this line here:
https://github.com/returntocorp/semgrep/blob/b031ff9a55cdc380fee335e0424384f0d1e4beff/src/osemgrep/cli_scan/Scan_subcommand.ml#L338-L340

And the the `Metrics_.send` after that.

Reply:
long term we want to support all command. Actually @hannesm recently improved significantly 'osemgrep ci', so we should just imitate what pysemgrep is doing.

Reply:
Ok, I will write a big `TODO`